keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java

```diff
@@ -260,64 +269,99 @@ private void closeAwsSecretsManagers() {
+      final CreateSecretRequest secretRequest =
+          CreateSecretRequest.builder()
+              .name(testSecretName)
+              .secretString(secretValue)
+              .tags(testSecretTag)
+              .build();
+      testSecretsManagerClient.createSecret(secretRequest).get(30, TimeUnit.SECONDS);
+      waitUntilSecretAvailable(testSecretName);
+      return testSecretName;
+    } catch (final Exception e) {
+      throw new RuntimeException(e.getMessage());
     }
   }
 
-  private void createTestSecrets() throws Exception {
-    secretName1WithTagKey1ValA = createTestSecret("tagKey1", "tagValA", SECRET_VALUE1);
-    secretName2WithTagKey1ValB = createTestSecret("tagKey1", "tagValB", SECRET_VALUE2);
-    secretName3WithTagKey2ValC = createTestSecret("tagKey2", "tagValC", SECRET_VALUE3);
-    secretName4WithTagKey2ValB = createTestSecret("tagKey2", "tagValB", SECRET_VALUE4);
-    testSecretNames = new ArrayList<>();
-    testSecretNames.addAll(
-        List.of(
-            secretName1WithTagKey1ValA,
-            secretName2WithTagKey1ValB,
-            secretName3WithTagKey2ValC,
-            secretName4WithTagKey2ValB));
+  private boolean checkTestSecretExistsAndUpdateIfUnmatched(
+      final String testSecretName, final Tag tag, final String secretValue) {
+    try {
+      final GetSecretValueResponse getSecretValueResponse =
+          testSecretsManagerClient
+              .getSecretValue(GetSecretValueRequest.builder().secretId(testSecretName).build())
+              .get(30, TimeUnit.SECONDS);
+      final DescribeSecretResponse describeSecretResponse =
+          testSecretsManagerClient
+              .describeSecret(DescribeSecretRequest.builder().secretId(testSecretName).build())
+              .get(30, TimeUnit.SECONDS);
+      if (!getSecretValueResponse.secretString().equals(secretValue)
+          && describeSecretResponse.hasTags()
+          && !describeSecretResponse.tags().get(0).equals(tag)) {
+        updateSecret(testSecretName, secretValue, describeSecretResponse.tags().get(0), tag);
+      }
+      return true;
+    } catch (final ResourceNotFoundException e) {
+      return false;
+    } catch (final Exception e) {
+      throw new RuntimeException(e.getMessage());
+    }
   }
 
-  private void waitUntilSecretAvailable(final String secretName)
-      throws ExecutionException, InterruptedException, TimeoutException {
-    testSecretsManagerClient
-        .getSecretValue(GetSecretValueRequest.builder().secretId(secretName).build())
-        .get(30, TimeUnit.SECONDS);
+  private void updateSecret(
+      final String testSecretName, final String secretValue, final Tag oldTag, final Tag newTag)
+      throws RuntimeException {
+    try {
+      testSecretsManagerClient
+          .updateSecret(
+              UpdateSecretRequest.builder()
+                  .secretId(testSecretName)
+                  .secretString(secretValue)
+                  .build())
+          .get(30, TimeUnit.SECONDS);
```

Comment:
Is there a delay between updating the secret and the new value being reflected in the GetSecretValueResponse?

Wondering if we'd need something like waitUntilSecretAvailable here...hopefully not, but might be worth considering if this turns out to be flaky again.